compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/VarargLowering.kt

```diff
@@ -102,7 +103,7 @@ private class VarargLowering(val context: JvmBackendContext) : FileLoweringPass,
         context.createJvmIrBuilder(currentScope!!.scope.scopeOwnerSymbol, startOffset, endOffset)
 
     private val IrFunctionSymbol.isArrayOf: Boolean
-        get() = this == context.ir.symbols.arrayOf || owner.isPrimitiveArrayOf
+        get() = this == context.ir.symbols.arrayOf || owner.isArrayOf
```

Comment:
Looks like it is already checking is this is `kotlin.Array`, so it is no need to check it by name

Reply:
Fir doesn't use the arrayOf symbol and therefore needs check by name.

Reply:
But then isn't the first condition (`this == context.ir.symbols.arrayOf`) unnecessary anymore?

Reply:
removed the first condition